flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
+    valf: np.ndarray(N,)
+        Function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+    const2: float
+        Constant C2
+
+    Raises
+    ------
+    ValueError
+        If C2 is not between 0 and 1
+    TypeError
+        If gradient function if not a function
+
+    Returns
+    -------
+    Boolean
+        True if curvature condition is satisfied
+    """
+    # Check parameters
+    if not isinstance(valf, (np.ndarray, float)):
+        raise TypeError('Variable should be an array of float')
+    if not isinstance(direction, (np.ndarray, float)):
+       raise TypeError('Descent direction should be an array of float')
+    if not isinstance(alpha, float):
+        raise TypeError("Alpha should be a float")
```

Comment:
Sorry to nitpick, but single quote? (This isn't really a PEP8 thing or anything... but it does change from single quote to double quote here though)